src/new/wallet.js

```diff
@@ -953,16 +1048,43 @@ class HathorWallet extends EventEmitter {
           continue;
         }
 
-        // If output was already used, we can't use it
+        const ret = {tx_id, index, address: output.decoded.address};
+        // If output was already used, we can't use it, unless requested in options
         if (output.spent_by) {
-          continue;
+          if (skipSpent) {
+            continue;
+          } else {
+            if (many) {
+              // If many we push to the array to be returned later
+              utxos.push(ret);
+            } else {
+              return [ret];
+            }
+          }
         }
 
         if (filterUTXOs(output)) {
-          return {tx_id, index, address: output.decoded.address};
+          if (many) {
+            // If many we push to the array to be returned later
+            utxos.push(ret);
+          } else {
+            return [ret];
+          }
         }
       }
     }
+
+    if (many) {
+      return utxos;
+    }
+  }
+
+  getMintAuthority(tokenUid, options = { many: false, skipSpent: true }) {
+    return this.selectAuthorityUtxo(tokenUid, wallet.isMintOutput.bind(wallet), options);
+  }
+
+  getMeltAuthority(tokenUid, options = { many: false, skipSpent: true }) {
```

Comment:
I fell you should let options be undefined. I see that the defaults here are the same as in `this.selectAuthorityUtxo` but weird behaviors might happen if the defaults are different.